content/doc/book/managing/cli.adoc

```diff
@@ -204,27 +196,78 @@ The general syntax for invoking the client is as follows:
 The `JENKINS_URL` can be specified via the environment variable `$JENKINS_URL`.
+Summaries of other general options can be displayed by running the client with no arguments at all.
+
+=== Client connection modes
+
+There are three basic modes in which the 2.54+ client may be used, selectable by global option:
+`-http`; `-ssh`; and `-remoting`.
+
+==== HTTP connection mode
+
+This is the default mode as of 2.54, though you may pass the `-http` option explicitly for clarity.
+
+Authentication is preferably with an `-auth` option, which takes a `username:apitoken` argument.
+Get your API token from `/me/configure`:
+
+[source]
+----
+java -jar jenkins-cli.jar [-s JENKINS_URL] -auth kohsuke:abc1234ffe4a command ...
+----
+
+(Actual passwords are also accepted, but this is discouraged.)
+
+You can also precede the argument with `@` to load the same content from a file:
+
+[source]
+----
+java -jar jenkins-cli.jar [-s JENKINS_URL] -auth @/home/kohsuke/.jenkins-cli command ...
+----
+
+==== SSH connection mode
+
+Authentication is via SSH keypair.
+You must select the Jenkins user ID as well:
+
+[source]
+----
+java -jar jenkins-cli.jar [-s JENKINS_URL] -ssh -user kohsuke command ...
+----
+
+In this mode, the client acts essentially like a native `ssh` command.
+
+==== Remoting connection mode
+
+This was the only mode supported by clients downloaded from a pre-2.54 Jenkins server
+(prior to the introduction of the `-remoting` option).
+Its use is deprecated for security and performance reasons.
+That said, certain commands or command modes can _only_ run in Remoting mode,
+typically because the command functionality involves running server-supplied code on the client machine.
 
-[TIP]
-====
-The `JENKINS_URL` environment variable is automatically set when Jenkins forks a process
-during builds or Pipelines, allowing the use of the Jenkins CLI from inside a
-project without explicit configuration of the Jenkins URL.
-====
+Authentication is preferably via SSH keypair.
+A `login` command and `--username` / `--password` command (note: *not global*) options are also available;
+these are discouraged since they cannot work with a non-password-based security realm,
+certain command arguments will not be properly parsed if anonymous users lack overall or job read access,
+and saving human-chosen passwords for use in scripts is considered insecure.
 
+Note that there are two transports available for this mode: over HTTP, or over a dedicated TCP socket.
+If the <<security#jnlp-tcp-port, TCP port is enabled>> and seems to work, the client will use this transport.
+If the TCP port is disabled, or such a port is advertised but does not accept connections
+(for example because you are using an HTTP reverse proxy with a firewall),
+the client will automatically fall back to the less efficient HTTP transport.
 
-=== Common Problems
+===== Common Problems with the Remoting-based client
 
 There are a number of common problems that may be experienced when running the
```

Comment:
This is far from an exhaustive list. In the links section of [JENKINS-41745](https://issues.jenkins-ci.org/browse/JENKINS-41745) I have tried to dig up various issues that I suspect are limited to Remoting mode.